frontend/src/components/readsets/ReadsetsListContent.tsx

```diff
@@ -1,105 +1,279 @@
-import React, { useEffect, useReducer } from "react"
+import React, { useCallback, useEffect, useMemo, useState } from "react"
 import AppPageHeader from "../AppPageHeader"
 import PageContent from "../PageContent"
 import ReadsetTableActions from "../../modules/readsetsTable/actions"
-import PagedItemsTable from "../pagedItemsTable/PagedItemsTable";
+import PagedItemsTable, { PagedItemsActionsCallbacks } from "../pagedItemsTable/PagedItemsTable";
 import FilterPanel from "../filters/filterPanel/FilterPanel";
 import FiltersBar from "../filters/filtersBar/FiltersBar";
 import { useAppDispatch, useAppSelector } from "../../hooks";
 import { selectReadsetsByID, selectReadsetsTable } from "../../selectors";
-import { BLOCKED, ObjectWithReadset, READSET_COLUMN_DEFINITIONS, READSET_COLUMN_FILTERS, READSET_FILTER_KEYS, RELEASED, ReadsetColumnID } from "./ReadsetsTableColumns";
+import { ObjectWithReadset, useReadsetColumnDefinitions, READSET_COLUMN_FILTERS, READSET_FILTER_KEYS, ReadsetColumnID } from "./ReadsetsTableColumns";
 import { Dataset, Readset } from "../../models/frontend_models";
 import { usePagedItemsActionsCallbacks } from "../pagedItemsTable/usePagedItemsActionCallbacks";
 import { useFilteredColumns } from "../pagedItemsTable/useFilteredColumns";
 import { useItemsByIDToDataObjects } from '../pagedItemsTable/useItemsByIDToDataObjects'
-import { Button, Tooltip } from "antd";
+import { Button, Spin, Tooltip } from "antd";
 import { ValidationStatus } from "../../modules/experimentRunLanes/models";
 import { MinusCircleTwoTone, PlusCircleTwoTone } from "@ant-design/icons";
-import { createFixedFilter } from "../../models/paged_items";
+import { createFixedFilter, FilterSet } from "../../models/paged_items";
 import { FILTER_TYPE } from "../../constants";
 import { setColumnWidths } from "../pagedItemsTable/tableColumnUtilities";
-import { setReleaseStatusAll } from "../../modules/datasets/actions";
-import { setReleaseStatus } from "../../modules/readsets/actions";
+import { ReleaseStatus } from "../../models/fms_api_models";
+import { ExpandableConfig } from "antd/lib/table/interface";
+import api from "../../utils/api";
+import produce from "immer";
+import { useCurrentUser } from "../../hooks/useCurrentUser";
+
+const RELEASE_STATUS_STRING = {
+    [ReleaseStatus.RELEASED]: 'Released',
+    [ReleaseStatus.BLOCKED]: 'Blocked',
+    [ReleaseStatus.AVAILABLE]: 'Available',
+} as const
+const OPPOSITE_STATUS = {
+    [ReleaseStatus.AVAILABLE]: ReleaseStatus.RELEASED,
+    [ReleaseStatus.RELEASED]: ReleaseStatus.BLOCKED,
+    [ReleaseStatus.BLOCKED]: ReleaseStatus.RELEASED,
+} as const
```

Comment:
i dont understand the "as const" at the end of these const defined variables.

Reply:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions

Reply:
yea, you ended up creating a readonly enum by typing it const and asserting it.  thats what i gathered, asserting makes sense but creating an enum also does